### ISSUE TITLE
Error 504: Can't update the balance and can't send coins

### ISSUE THREAD

**Issue by JasonG-FR** [garlicoin-project/copay-grlc#5]
Hi,

I'm having an issue with the Android App, it seems that it can't reach the server.
Here are two screenshots showing the issue. My Internet access is working.

![screenshot_20180222-222754](https://user-images.githubusercontent.com/9202329/36565603-3fae38fc-1821-11e8-93a7-fd2c6f2cc703.png)
![screenshot_20180222-223554](https://user-images.githubusercontent.com/9202329/36565604-3fdbf760-1821-11e8-81fd-8ca5ac5ad5fc.png)

I'm joining the session log too : 
`Garlicoin Wallet v3.2.8 #dc278d5
Client: {"isAndroid":true,"isIOS":false,"isWP":false,"isSafari":false,"ua":"Mozilla/5.0 (Linux; Android 7.1.2; A0001 Build/NJH47F; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/63.0.3239.111 Mobile Safari/537.36","isCordova":true,"isMobile":true,"isDevel":false,"supportsTrezor":false,"versionIntelTEE":"","supportsIntelTEE":false}
Using: FileStorage
Running head controller:Garlicoin Wallet
Route change from: - to: starting
toParams:{}
fromParams:{}
Starting...
Init profile...
Got main dir: file:///data/user/0/com.garlicinsight.garlicwallet/files/
Profile is not encrypted
Profile read
Preferences read: {"limits":{"totalCopayers":6,"mPlusN":100},"bws":{"url":"https://gws.garlicinsight.com/gws/api"},"download":{"copay":{"url":"https://garlicoin.io"},"bitpay":{"url":"https://play.google.com/store/apps/details?id=com.garlicinsight.garlicwallet"}},"rateApp":{"copay":{"ios":"https://garlicoin.io","android":"https://garlicoin.io","wp":""},"bitpay":{"ios":"https://garlicoin.io","android":"https://play.google.com/store/apps/details?id=com.garlicinsight.garlicwallet","wp":""}},"wallet":{"requiredCopayers":2,"totalCopayers":3,"spendUnconfirmed":false,"reconnectDelay":5000,"idleDurationMin":4,"settings":{"unitName":"GRLC","unitToSatoshi":100000000,"unitDecimals":8,"unitCode":"grlc","alternativeName":"US Dollar","alternativeIsoCode":"USD","defaultLanguage":"en","feeLevel":"normal"}},"lock":{"method":"pin","value":"0576","bannedUntil":null},"recentTransactions":{"enabled":true},"hideNextSteps":{"enabled":true},"rates":{"url":"https://garlicinsight.com:443/api/rates"},"release":{"url":"https://api.github.com/repos/garlicoin-project/copay-grlc/releases/latest"},"pushNotificationsEnabled":true,"emailNotifications":{"enabled":false},"colorFor":{"681b94c6-1253-4e66-9339-ac4632a7fecf":"#4a90e2","ea73a9bd-4a73-4dff-b2bd-a8547651684f":"#4a90e2"},"bwsFor":{},"aliasFor":{},"emailFor":{}}
Preferences read
Binding wallet:ea73a9bd-4a73-4dff-b2bd-a8547651684f Validating?:false
Wallet + ea73a9bd-4a73-4dff-b2bd-a8547651684f status:undefined
Bound 1 out of 1 wallets
Profile loaded ... Starting UX.
Trying to pre-initialize QRScanner.
Setting default language: en
Route change from: starting to: tabs.home
toParams:{"fromOnboarding":""}
fromParams:{}
scannerService is reviewing platform capabilities...
A camera is available to this app.
Camera access is authorized.
Support for opening device settings is available on this platform.
A light is available on this platform.
A second camera is available on this platform.
Camera permission already granted.
Initializing scanner...
scannerService is reviewing platform capabilities...
A camera is available to this app.
Camera access is authorized.
Support for opening device settings is available on this platform.
A light is available on this platform.
A second camera is available on this platform.
donate Service, updating nextSteps. linked/total: 0/0
Adding NextSteps entry:donate
Updating Status: Personal Wallet null
Starting push notification registration...
startupService is showing the StatusBar...
startupService is hiding the splashscreen...
Get token for push notifications: fIBsXRpDFm4:APA91bEtvtlQGqTDqldh6xxWrPZJJ5yWLmdi8sVNgvBCFbVsc6BqfmTR50gHBLBb6kdyFUW-Cb4tP6gRLGrHc97Ebvp8lCcoka7WvLblmYySUC1-ny01YgeAehDtzeN-BmBdOzgYCi6Q
Initializing openURL
Route change from: tabs.home to: tabs.wallet
toParams:{"walletId":"ea73a9bd-4a73-4dff-b2bd-a8547651684f","fromOnboarding":""}
fromParams:{"fromOnboarding":""}
Updating Status: Personal Wallet null
Updating Transaction History
Route change from: tabs.wallet to: tabs.home
toParams:{"fromOnboarding":""}
fromParams:{"walletId":"ea73a9bd-4a73-4dff-b2bd-a8547651684f","fromOnboarding":""}
Updating Status: Personal Wallet null
Route change from: tabs.home to: tabs.settings
toParams:{}
fromParams:{"fromOnboarding":""}
Route change from: tabs.settings to: tabs.advanced
toParams:{}
fromParams:{}
Route change from: tabs.advanced to: tabs.settings
toParams:{}
fromParams:{}
Route change from: tabs.settings to: tabs.about
toParams:{}
fromParams:{}
Route change from: tabs.about to: tabs.about.logs
toParams:{}
fromParams:{}
Personal Wallet: Subscription Push Notifications success.
Status update. hash:385339920 Try:0
Got Wallet Status for:Personal Wallet
Unknown error type: 504
504`

**Comment by JasonG-FR** [comment]
It's working again now, maybe the server wasn't reachable ?

**Comment by Tonimit** [comment]
I have also problems sending coins.
Here is what msg I got:
![garlig5](https://user-images.githubusercontent.com/36791665/36630485-4e7448da-1967-11e8-880e-3e3dccb11087.jpg)
![garlig6](https://user-images.githubusercontent.com/36791665/36630486-53251d28-1967-11e8-8c3d-b84623dc9d43.jpg)
any solution?

**Comment by ryan-shaw** [comment]
Before the error occurs how long does it hang for during sending?

**Comment by ryan-shaw** [comment]
Also I notice the fee is very large, are you mining into this wallet? If
so, it could be an issue with combining a large number of utxo’s can you
try sending a smaller amount?

**Comment by Tonimit** [comment]
Hi Ryan,

I've tried to send 10 coins and it's working fine.
Yes, I'm mining to this wallet.
How big transfers I should make?

thank you.

**Comment by ryan-shaw** [comment]
Yeah it’s what I thought then. I’m not sure how big the transfers should be
but you can experiment, what’s happening is the request is taking longer
than the gateway timeout causing the request to fail. I can up the timeout
to help with bigger txs, but for now could you send the funds with smaller
txs until I can fix it.

Thanks

**Comment by Tonimit** [comment]
Thanks!
